app/uk/gov/hmrc/agentclientrelationships/connectors/HIPConnector.scala

```diff
@@ -0,0 +1,418 @@
+    val encodedClientId = UriEncoding.encodePathSegment(taxIdentifier.value, "UTF-8")
+    taxIdentifier match {
+      case MtdItId(_) =>
+        new URL(
+          s"$baseUrl/RESTAdapter/rosm/agent-relationship?refNumber=$encodedClientId&isAnAgent=false&activeOnly=true&regime=${getRegimeFor(taxIdentifier)}&relationshipType=ZA01&authProfile=ALL00001"
+        )
+      case Vrn(_) =>
+        new URL(
+          s"$baseUrl/RESTAdapter/rosm/agent-relationship?idType=VRN&refNumber=$encodedClientId&isAnAgent=false&activeOnly=true&regime=${getRegimeFor(taxIdentifier)}&relationshipType=ZA01&authProfile=ALL00001"
+        )
+      case Utr(_) =>
+        new URL(
+          s"$baseUrl/RESTAdapter/rosm/agent-relationship?idType=UTR&refNumber=$encodedClientId&isAnAgent=false&activeOnly=true&regime=${getRegimeFor(taxIdentifier)}"
+        )
+      case Urn(_) =>
+        new URL(
+          s"$baseUrl/RESTAdapter/rosm/agent-relationship?idType=URN&refNumber=$encodedClientId&isAnAgent=false&activeOnly=true&regime=${getRegimeFor(taxIdentifier)}"
+        )
+      case CgtRef(_) =>
+        new URL(
+          s"$baseUrl/RESTAdapter/rosm/agent-relationship?idType=ZCGT&refNumber=$encodedClientId&isAnAgent=false&activeOnly=true&regime=${getRegimeFor(taxIdentifier)}&relationshipType=ZA01&authProfile=ALL00001"
+        )
+      case PptRef(_) =>
+        new URL(
+          s"$baseUrl/RESTAdapter/rosm/agent-relationship?idType=ZPPT&refNumber=$encodedClientId&isAnAgent=false&activeOnly=true&regime=${getRegimeFor(taxIdentifier)}&relationshipType=ZA01&authProfile=ALL00001"
+        )
+      case CbcId(_) =>
+        new URL(
+          s"$baseUrl/RESTAdapter/rosm/agent-relationship?idType=CBC&refNumber=$encodedClientId&isAnAgent=false&activeOnly=true&regime=${getRegimeFor(taxIdentifier)}"
+        )
+      case PlrId(_) =>
+        new URL(
+          s"$baseUrl/RESTAdapter/rosm/agent-relationship?idType=ZPLR&refNumber=$encodedClientId&isAnAgent=false&activeOnly=true&regime=${getRegimeFor(taxIdentifier)}"
+        )
+      case _ => throw new IllegalStateException(s"Unsupported Identifier $taxIdentifier")
+
+    }
+  }
+
+  private def getRegimeFor(clientId: TaxIdentifier): String =
+    clientId match {
+      case MtdItId(_) => "ITSA"
+      case Vrn(_)     => "VATC"
+      case Utr(_)     => "TRS"
+      case Urn(_)     => "TRS"
+      case CgtRef(_)  => "CGT"
+      case PptRef(_)  => "PPT"
+      case CbcId(_)   => "CBC"
+      case PlrId(_)   => "PLR"
+      case _          => throw new IllegalArgumentException(s"Tax identifier not supported $clientId")
+    }
+
+  private def createAgentRelationshipHipInputJson(enrolmentKey: EnrolmentKey, arn: String) =
+    includeIdTypeIfNeeded(enrolmentKey)(
+      Json
+        .parse(s"""{
+          "arn": "$arn",
+          "refNumber": "${enrolmentKey.oneIdentifier().value}",
+          "regime": "${getRegimeFor(enrolmentKey.oneTaxIdentifier())}",
+          "action": "0001",
+          "isExclusiveAgent": true
```

Comment:
we must only send with true if the service is not HMRC-MTD-IT-SUPP. Setting this to true will tell ETMP to de-auth all other agents for the service; we don't want that for SUPP